### PR TITLE
Add retries to flaky multiroot semaphores tests.

### PR DESCRIPTION
**Pull request by Mati365**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Tests: Add `retry` option to flaky `multiroot` test, extend delay between assertions. 
Tests: Randomize the order of the tests, to prevent incorrect dependencies across tests. 

---

### Additional information

This test tests quite rare scenario when `setData` is being called when editor is not initialized. In theory - it is possible, but I haven't managed to reproduce this in the real world. At this moment, there were no reports about incorrect behavior, so it's safe enough to retry it a few times.

### CODE REVIEW

File: tests/useMultiRootEditor.test.tsx [hunk 1]
```diff
@@ -896,7 +896,7 @@ describe( 'useMultiRootEditor', () => {
 
 	describe( 'semaphores', () => {
 		const testSemaphoreForWatchdog = enableWatchdog => {
-			it( 'should assign properly `data` property to editor even if it is still mounting', async () => {
+			it( 'should assign properly `data` property to editor even if it is still mounting', { retry: 3 }, async () => {
```
Comment (gorzelinski) [inline tests/useMultiRootEditor.test.tsx:899]:
```suggestion
			it( 'should assign `data` property to the editor even if it is still mounting', { retry: 3 }, async () => {
```

File: tests/useMultiRootEditor.test.tsx [hunk 2]
```diff
@@ -931,16 +931,20 @@ describe( 'useMultiRootEditor', () => {
 					editor: SlowEditor
 				} ) );
 
-				await timeout( 100 );
+				await timeout( 500 );
 
+				// Depending on the execution order on the event loop this `setData` might be delayed by the React engine.
```
Comment (gorzelinski) [inline tests/useMultiRootEditor.test.tsx:936]:
```suggestion
				// Depending on the execution order on the event loop, this `setData` might be delayed by the React engine.
```

File: tests/useMultiRootEditor.test.tsx [hunk 3]
```diff
@@ -931,16 +931,20 @@ describe( 'useMultiRootEditor', () => {
...
+				// It happens only if event loop is busy and React has to wait for the next tick a little bit longer than usual.
```
Comment (gorzelinski) [inline tests/useMultiRootEditor.test.tsx:937]:
```suggestion
				// It happens only if event loop is busy and React has to wait for the next tick a little bit longer than usual.
```
```suggestion
				// It happens only if the event loop is busy and React has to wait for the next tick a little bit longer than usual.
```

File: tests/useMultiRootEditor.test.tsx [hunk 4]
```diff
@@ -931,16 +931,20 @@ describe( 'useMultiRootEditor', () => {
...
+				// It does not play well with `waitFor` below so we added few retries to make it more stable.
```
Comment (gorzelinski) [inline tests/useMultiRootEditor.test.tsx:938]:
```suggestion
				// It does not play well with the `waitFor` below, so we added a few retries to make it more stable.
```

File: tests/useMultiRootEditor.test.tsx [hunk 5]
```diff
@@ -931,16 +931,20 @@ describe( 'useMultiRootEditor', () => {
...
+				// It should not be a problem in real life, as it is a rare case and might be solved in future React versions.
 				result.current.setData( {
 					intro: 'Hello World!',
 					content: ''
 				} );
 
-				await timeout( 200 );
+				await timeout( 500 );
```
Comment (gorzelinski) [inline tests/useMultiRootEditor.test.tsx:945]: It's not super elegant, but I understand the problems with React.

Comment (Mati365) [reply tests/useMultiRootEditor.test.tsx:945]: Yep :[

**Review by gorzelinski — APPROVED** [review]

**Pull request merged**